### PR TITLE
Add Python based builder image

### PR DESCRIPTION
**Pull request by JoelSpeed**
Adds a Python 3.7 based builder image for linting and testing python based projects

### CODE REVIEW

File: images/python-builder/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+		make \
+		patch \
+		unzip \
+		xz-utils \
+		zlib1g-dev \
+		tk-dev \
+		uuid-dev \
+	&& rm -rf /var/lib/apt/lists/*
+
+ENV GPG_KEY 0D96DF4D4110E5C43FBFB17F2D347EA6AA65421D
```
Comment (icelynjennings) [inline images/python-builder/Dockerfile]: What's this?

**Review by icelynjennings — APPROVED** [review]

File: images/python-builder/Dockerfile (hunk 1 above)
Comment (JoelSpeed) [reply images/python-builder/Dockerfile]: Copy-pasta https://github.com/docker-library/python/blob/65b5603fdf177c6d90df6ea6f3bd6e23839c63a4/3.7/stretch/Dockerfile#L9-L88

File: images/README.md [hunk 2]
```diff
@@ -47,3 +47,13 @@ The Kubebuilder builder contains the following:
   - kube-apiserver
   - kubectl
   - kubebuilder
+
+## Python Builder
+
+The Python builder (`quay.io/pusher/python-builder`) is for testing Python based
+projects within Prow.
+
+The Python builder conaints the following:
```
Comment (icelynjennings) [inline images/README.md]:
```suggestion
The Python builder contains the following:
```

**Review by icelynjennings — CHANGES_REQUESTED** [review]

File: images/README.md (hunk 2 above)
Comment (JoelSpeed) [reply images/README.md]: Good spot 👌

**Review by icelynjennings — APPROVED** [review]

**Pull request merged**